for_each loop fails to parse namespaced identifiers in collection position

## Summary

When using a namespaced identifier (e.g., `lib.HeroClassStr`) as the collection in a `for_each` loop, the parser fails with unexpected errors. This occurs when a module is imported but not used (i.e., `import lib"..."` without `& use`), requiring the namespace prefix for all public symbols.

## Reproduction

**File:** `EZ-TAG/main/main.ez` (line 55)

```ez
import & use @std, @strings
import lib"../src/library"

do get_hero_class() -> lib.HeroClass {
    temp class lib.HeroClass
    temp count int = 0
    for_each c in lib.HeroClassStr {  // <-- BUG: fails at build time
        count += 1
        println("${count}.${c}")
    }
    // ...
}
```

**Library definition:** `EZ-TAG/src/library/types.ez`
```ez
module library

const HeroClass enum {
    WARRIOR
    MAGE
}

const HeroClassStr [string, 2] = {"Warrior", "Mage"}
```

## Expected Behavior

The `for_each` loop should correctly parse `lib.HeroClassStr` as the collection to iterate over, since:
- `lib.HeroClass` works correctly as a return type (line 51)
- `lib.HeroClass` works correctly in variable declarations (line 52)
- `lib.HeroClass.WARRIOR` works correctly in assignments (line 65)

## Actual Behavior

Build-time parser errors:

```
error[E2002]: expected :, got += instead
  --> EZ-TAG/main/main.ez:57:16
   |
57 |         count += 1
   |                ^^ expected token not found

error[E2002]: expected {, got += instead
  --> EZ-TAG/main/main.ez:57:16

error[E2001]: unexpected token '+='
  --> EZ-TAG/main/main.ez:57:16

error[E2003]: expected expression, found end of block
  --> EZ-TAG/main/main.ez:77:1
```

## Additional Notes

- Removing the `lib.` prefix (using just `HeroClassStr`) allows the code to compile but fails at runtime (identifier not found)
- The parser appears to stop parsing the expression at `lib` and doesn't continue to parse the `.HeroClassStr` member access
- Other contexts (return types, variable declarations, enum member access) handle namespaced identifiers correctly

## Affected Code

Likely in `pkg/parser/parser.go` in the `parseForEachStatement()` function around line 1356:
```go
stmt.Collection = p.parseExpression(LOWEST)
```

The expression parsing may be terminating early or not properly handling the DOT token in this context.

--- a/integration-tests/pass/multi-file/foreach-namespaced-collection/main.ez
+++ b/integration-tests/pass/multi-file/foreach-namespaced-collection/main.ez
@@ -1,0 +1,55 @@
+/*
+ * Test for_each with namespaced identifier as collection
+ * Regression test for bug #534
+ *
+ * This tests that for_each correctly parses lib.Items as a collection
+ * when an import uses namespace prefix (import lib"..." without & use).
+ */
+import @std
+import lib"./testlib"
+
+using std
+
+do main() {
+    println("=== For_each Namespaced Collection Test ===")
+    temp passed int = 0
+    temp failed int = 0
+
+    // Test 1: for_each over namespaced string array
+    println("Items from lib.Items:")
+    temp item_count int = 0
+    for_each item in lib.Items {
+        println("  ${item}")
+        item_count += 1
+    }
+    if item_count == 3 {
+        println("  [PASS] for_each over namespaced string array")
+        passed += 1
+    } otherwise {
+        println("  [FAIL] expected 3 items, got ${item_count}")
+        failed += 1
+    }
+
+    // Test 2: for_each over namespaced int array with accumulation
+    temp sum int = 0
+    for_each n in lib.Numbers {
+        sum += n
+    }
+    if sum == 100 {
+        println("  [PASS] for_each over namespaced int array (sum=${sum})")
+        passed += 1
+    } otherwise {
+        println("  [FAIL] for_each over namespaced int array: expected 100, got ${sum}")
+        failed += 1
+    }
+
+    // Summary
+    println("")
+    println("Results: ${passed} passed, ${failed} failed")
+
+    if failed > 0 {
+        println("SOME TESTS FAILED")
+    } otherwise {
+        println("ALL TESTS PASSED")
+    }
+}

--- a/integration-tests/pass/multi-file/foreach-namespaced-collection/testlib/lib.ez
+++ b/integration-tests/pass/multi-file/foreach-namespaced-collection/testlib/lib.ez
@@ -1,0 +1,4 @@
+module testlib
+
+const Items [string, 3] = {"apple", "banana", "cherry"}
+const Numbers [int, 4] = {10, 20, 30, 40}

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -2957,12 +2957,16 @@ func (p *Parser) parseMemberExpression(left Expression) Expression {
 			if len(ident.Value) > 0 && ident.Value[0] >= 'A' && ident.Value[0] <= 'Z' {
 				return exp // It's an enum access, not a struct literal
 			}
-			qualifiedName := &Label{
-				Token: exp.Token,
-				Value: ident.Value + "." + memberName,
+			// Use speculative lookahead to verify this is actually a struct literal
+			// and not a block statement (e.g., in for_each item in lib.Items { ... })
+			if p.looksLikeStructLiteral() {
+				qualifiedName := &Label{
+					Token: exp.Token,
+					Value: ident.Value + "." + memberName,
+				}
+				p.nextToken() // move to {
+				return p.parseStructLiteralFromIdent(qualifiedName)
 			}
-			p.nextToken() // move to {
-			return p.parseStructLiteralFromIdent(qualifiedName)
 		}
 	}
 

